Sources/CartesianComponentsRepresentable.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  CartesianComponentsRepresentable.swift
+//  Euclid
+//
+//  Created by Ioannis Kaliakatsos on 29.11.2020.
+//  Copyright © 2020 Nick Lockwood. All rights reserved.
+//
+
+import Foundation
+
+private enum CodingKeys: CodingKey {
+    case x, y, z
+}
+
+public protocol CartesianComponentsRepresentable: Codable, Hashable {
+    var x: Double { get }
+    var y: Double { get }
+    var z: Double { get }
+    init(x: Double, y: Double, z: Double)
+}
+
+public extension CartesianComponentsRepresentable {
+    var norm: Double {
```

Comment:
Why `norm` and not `length` or `magnitude`?

Reply:
My thinking was to use mathematical terminology as much as possible, in order to be concise. So, _vector norm_ was my choice.